raiden/transfer/utils.py

```diff
@@ -53,3 +58,12 @@ def is_valid_secret_reveal(
     transfer_secrethash: SecretHash,
 ) -> bool:
     return state_change.secrethash == transfer_secrethash
+
+
+def encrypt_secret(secret: Secret, target_metadata: Optional[AddressMetadata]) -> EncryptedSecret:
+    encrypted_secret = EncryptedSecret(b"")
+    target_metadata = target_metadata or {}
+    public_key = target_metadata.get("displayname")
```

Comment:
The ticket description states that:

```Initiator always knows target's public key, due to knowing their presence, which allows the pubkey to be derived from displayName/userId signature```

I wasn't sure how this should be done, so this is a placeholder for getting the actual public key of the target. What's the correct way to derive this key?

Reply:
> The ticket description states that:
> 
> `Initiator always knows target's public key, due to knowing their presence, which allows the pubkey to be derived from displayName/userId signature`
> 
> I wasn't sure how this should be done, so this is a placeholder for getting the actual public key of the target. What's the correct way to derive this key?

I think the key word in that description is _signature_. Namely, if you have a signature (of the displayName/userId in this case), you can recover the public key corresponding to the private key the message was signed with.
Take a look at https://github.com/raiden-network/raiden/blob/1214b7980c7f61f73ac94e476ff186a777bf10dc/raiden/utils/signer.py#L24-L40.